netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -221,6 +223,16 @@ public NettyChannelBuilder maxHeaderListSize(int maxHeaderListSize) {
     return this;
   }
 
+  final NettyChannelBuilder initHandler(ChannelHandler initHandler) {
+    if (initHandler != null) {
```

Comment:
nit: maybe we can disallow `null`s here. Or if we want to allow calling with `null` which does nothing, annotate with `@Nullable`

Reply:
For the open source case we usually dont use it, so the handler would serve only to remove itself.  Internally, we would set it.   I'd prefer to not auto remove itself for the common case.